### PR TITLE
base: Ostream helpers (iterable, tuple, pair, enum, pointers, optional)

### PR DESCRIPTION
**Pull request by MeatBoy106**
Refinements, tests and extra features for ostream helpers. Most of these imported and updated from Gerrit submission.

### CODE REVIEW

**Review by powerjg — DISMISSED** [review]

File: src/base/stl_helpers/ostream_helpers.test.cc
Comment (giactra) [inline src/base/stl_helpers/ostream_helpers.test.cc]: Nit: this should be lower case instead of lower camel case being a local variable

File: src/base/stl_helpers/ostream_helpers.test.cc
Comment (giactra) [inline src/base/stl_helpers/ostream_helpers.test.cc]: Ditto

File: src/base/stl_helpers/ostream_helpers.test.cc
Comment (giactra) [inline src/base/stl_helpers/ostream_helpers.test.cc]: Ditto

**Review by giactra — CHANGES_REQUESTED** [review]
Sorry to stop the merging process. Overall LGTM but I have a minor request

**Comment by BobbyRBruce** [comment]
Hey @MeatBoy106 : This isn't a harder requirement but it would help us a little bit if you could add tags to the PR title as well. We're still, for now, enforcing the use of tags and github, by default, uses the PR title as the merge commit header, so it's nice to have that setup for us (when a Maintainer squashes and merges the commit they can choose to edit the header, but that means they choose what the tags are. It's nice if the contributor can do that).

**Comment by MeatBoy106** [comment]
@BobbyRBruce Sure! Is it better that way?

**Comment by BobbyRBruce** [comment]
> @BobbyRBruce Sure! Is it better that way?

Yes. Thank you!

**Review by BobbyRBruce — APPROVED** [review]

**Comment by BobbyRBruce** [comment]
I'll merge this when @giactra has approved.

**Review by giactra — APPROVED** [review]
Thanks for the change

**Pull request merged**
